### ISSUE TITLE
Suggestion: multi-language without the need of inline documentation switches

### ISSUE THREAD

**Issue by remij1** [bartosz-antosik/vscode-spellright#171]
Hi ! I may have not seen this option, but it seems to me that there is currently no way to check a document with multiple languages at the same time.

As an example, I am currently writing a Latex document in French, but it has a lot of technical words which are written in English. It would be great if these could be checked too, without being marked as wrong.

**Comment by bartosz-antosik** [comment]
You are correct. Currently it is not possible to check every token in the document in multiple languages. 

It is possible to have sections of the document spelled in various languages (In-Document commands can switch) or syntactical parts (e.g. comments in one language, body in another).

I will think on this idea.

**Comment by GandalfSaxe** [comment]
I'd also like to see this. My use-case is a work log used by multiple people on a small team, where some prefers Danish, some prefers English, but all are proficient in both and just use whatever language they prefer in the moment.

It may sound horrible, but it works if it's just an internal document.

**Comment by Zyxyx** [comment]
I would really like this feature too for bilingual (EN-FR) files editing.
Thank you in advance...

**Comment by bartosz-antosik** [comment]
Me too! And I am thinking of this but it is somehow complicated at current state of affairs.

**Comment by GandalfSaxe** [comment]
Isn't it just having this option:

```
"spellright.language": "en"
```

allow for

```
"spellright.language": ["en", "fr"]
```

And effectively just apply both dictionaries?

**Comment by JingleChannel** [comment]
I would love to see multiple dictionary support as well! Our READMEs and business objects are in German, the rest of the development is done in English. This workflow might apply to other enterprises as well.

**Comment by bartosz-antosik** [comment]
@KonradDeskiewicz Do I get it correctly that false positives come from the fact that e.g. Comments & strings are in different languages? Or is it a mixture altogether?

**Comment by KonradDeskiewicz** [comment]
Mostly false positive is java script code, variable names etc.
like this:
![image](https://user-images.githubusercontent.com/17060750/45089198-b4bf6980-b10b-11e8-90a6-4306851ca727.png)

**Comment by bartosz-antosik** [comment]
@KonradDeskiewicz I doubt multiple languages would cure the problem entirely (see there still would be `params`, `laroute` or `axios`). As a temporary remedy you may have a look at `"spellright.ignoreRegExps"` setting which can skip `<script></script>` content from spelling.

**Comment by KonradDeskiewicz** [comment]
That's why a said "a lot" ;-)
Regex way cuts most of them but there are still variables within template part of `.vue` files. 

I wonder if you can use TextMate grammars `string.` to select strings same way as they are used for syntax highlighting.

**Comment by bartosz-antosik** [comment]
@KonradDeskiewicz ("I wonder if ...") That's the point! I would love if VSCode would expose syntactical information to extensions! Then spelling e.g. CamelCase variables would make sense. As far as I know there is no information about the syntax which is used for coloring accessible for extensions. And I am really unable to write parser for everything. There are extensions which do "brute force" approach but somehow I do not like this idea (would require providing & maintaining dozens of supplementary dictionaries e.g. language keywords etc.)

**Comment by bkircher** [comment]
Very cool and much appreciated. Can you push the code to a branch, too? Happy to test that branch then.

**Comment by bartosz-antosik** [comment]
Just released version 3.0.0 has the concurrent multiple language spelling functionality.
